var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -128,8 +129,49 @@ def autoreconf(self, spec, prefix):
             autoreconf('-iv')
 
     def configure_args(self):
-        args = ['--with-mpi=%s' % self.spec['mpi'].prefix,
-                'SEQ_CC=%s' % spack_cc]
+        # for convenience
+        spec = self.spec
+
+        # mvapich2 will set the MPI env variables,
+        # but that may not be a given. So set them for safety
+        args = ['--with-mpi=%s' % spec['mpi'].prefix,
+                'SEQ_CC=%s' % spack_cc,
+                'MPICC=%s'  % spec['mpi'].mpicc,
+                'MPICXX=%s' % spec['mpi'].mpicxx,
+                'MPIF77=%s' % spec['mpi'].mpifc,
+                'MPIF90=%s' % spec['mpi'].mpifc,
+                ]
+
+        # setup testing (required at configure time)
+        mpiexec = ''
+        mpiexec_flags = ''
+        # this will be an ordered search.
+        # that isn't great - for example, some HPC machines may use
+        # slurm to manage jobs, but another process launcher to run
+        # e.g., Cray can use aprun + slurm, or mpirun + slurm.
+        mpiexecs = ['srun', 'aprun', 'jsrun', 'mpiexec', 'mpirun']
+        mpiexecs_flags = {'srun': '--cpu-bind=cores -c1 -n',
+                          'aprun': '-n',
+                          'jsrun': '-p',
+                          'mpiexec': '-n',
+                          'mpirun': 'np',
+                          }
+
+        for candidate_mpiexec in mpiexecs:
+            x = which(candidate_mpiexec)
+            if x:
+                mpiexec = x
```

Comment:
I think as-is it can be changed.  The logic originally set the variables afterwards, but the current organization ensures that the test variables are *not* set unless a valid launcher is found

Reply:
Fixed